cli/migrate.go

```diff
@@ -283,42 +321,31 @@ func publishEventWithOptions(ctx context_pkg.Context, envName string, modifiedSc
 		if err := env.HandleEvent(eventName, eventContext); err != nil {
 			log.Fatalf("Failed to handle event '%s': %s", eventName, err)
 		}
+
+		if err := migration.UnmarkSchema(s.ID); err != nil {
+			log.Fatalf("Failed to remove '%s' from pending post migration schemas: %s", s.ID, err)
+		}
 	}
 
+	clearDeadPendingSchemas(modifiedSchemas, manager)
+
 	if syncETCDEvent {
 		if _, err := server.NewSyncWriter(sync, db).Sync(ctx); err != nil {
 			log.Fatalf("Failed to synchronize post-migration events, err: %s", err)
 		}
 	}
 }
 
-func publishEvent(ctx context_pkg.Context, envName string, modifiedSchemas []string, eventName string, syncETCDEvent bool, eventTimeout time.Duration) error {
-	config := util.GetConfig()
-
-	rawDB, err := dbutil.CreateFromConfig(config)
-
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	db := server.NewDbSyncWrapper(rawDB)
-
-	sync, err := sync_util.CreateFromConfig(config)
-
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	ident, err := middleware.CreateIdentityServiceFromConfig(config)
+func clearDeadPendingSchemas(modifiedSchemas []string, manager *schema.Manager) {
+	for _, schemaID := range modifiedSchemas {
+		_, exists := manager.Schema(schemaID)
+		if exists {
+			continue
+		}
 
-	if err != nil {
-		log.Fatal(err)
+		log.Warning("Found no longer existing schema '%s' in pending post-migration events, removing", schemaID)
+		if err := migration.UnmarkSchema(schemaID); err != nil {
+			log.Fatalf("Failed to remove no longer existing '%s' from pending post migration schemas: %s", schemaID, err)
```

Comment:
will this error cause event to be executed again after restart?

Reply:
Yes.